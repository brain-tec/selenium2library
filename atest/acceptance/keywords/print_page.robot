*** Settings ***
Documentation    Suite description
Suite Setup       Go To Page "non_ascii.html"
Resource          ../resource.robot

Test Setup    Remove Files    ${OUTPUTDIR}/selenium-page-*.pdf

*** Test Cases ***
Print Page As PDF Without Print Options
    Print Page As PDF

Verify Index Increments With Multiple Prints
    [Setup]    Remove Files    ${OUTPUTDIR}/selenium-page-*.pdf
    ${file_1} =    Print Page As PDF    background=${True}  scale=${2}
    Should Be Equal    ${file_1}    ${OUTPUTDIR}${/}selenium-page-1.pdf
    ${file_2} =    Print Page As PDF    orientation=landscape
    Should Be Equal    ${file_2}    ${OUTPUTDIR}${/}selenium-page-2.pdf
    Go To  https://robotframework.org/foundation/
    ${file_3} =    Print Page As PDF    shrink_to_fit=${True}  page_height=${35.56}  page_width=${21.59}
    Should Be Equal    ${file_3}    ${OUTPUTDIR}${/}selenium-page-3.pdf

Print With Full Options
    Print Page As PDF  page_ranges=['1']  background=${False}  shrink_to_fit=${False}  orientation=portrait
    ...  margin_top=${0.5}  margin_left=${1.5}  margin_bottom=${0.5}  margin_right=${1.5}
    ...  page_height=${35.56}  page_width=${21.59}
