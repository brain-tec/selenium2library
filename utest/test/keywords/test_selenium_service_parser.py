import os
import sys
import unittest

import pytest
from approvaltests.approvals import verify_all
from approvaltests.reporters.generic_diff_reporter_factory import (
    GenericDiffReporterFactory,
)
from mockito import mock, when, unstub, ANY
from robot.utils import WINDOWS
from selenium import webdriver

from SeleniumLibrary.keywords.webdrivertools import SeleniumService, WebDriverCreator


@pytest.fixture(scope="module")
def service():
    return SeleniumService()

@pytest.fixture(scope="module")
def reporter():
    path = os.path.dirname(__file__)
    reporter_json = os.path.abspath(
        os.path.join(path, "..", "approvals_reporters.json")
    )
    factory = GenericDiffReporterFactory()
    factory.load(reporter_json)
    return factory.get_first_working()


def teardown_function():
    unstub()


@unittest.skipIf(WINDOWS, reason="ApprovalTest do not support different line feeds")
def test_parse_service_string(service, reporter):
    results = []
    results.append(service._parse('attribute="arg1"'))
    # results.append(service._parse("  attribute = True  "))    # need to resolve issues with spaces in service string.
    results.append(service._parse('attribute="arg1";attribute=True'))
    results.append(service._parse('attribute=["arg1","arg2","arg3"] ; attribute=True ; attribute="arg4"'))
    results.append(
        service._parse(
            'attribute="C:\\\\path\\to\\\\profile"'
        )
    )
    results.append(
        service._parse(
            r'attribute="arg1"; attribute="C:\\path\\to\\profile"'
        )
    )
    results.append(service._parse("attribute=None"))
    verify_all("Selenium service string to dict", results, reporter=reporter)


# @unittest.skipIf(WINDOWS, reason="ApprovalTest do not support different line feeds")
# @unittest.skipIf(sys.version_info > (3, 11), reason="Errors change with Python 3.12")
@pytest.mark.skipif(WINDOWS, reason="ApprovalTest do not support different line feeds")
@pytest.mark.skipif(sys.version_info > (3, 11), reason="Errors change with Python 3.12")
def test_parse_service_string_errors(service, reporter):
    results = []
    results.append(error_formatter(service._parse, "attribute=arg1", True))
    results.append(error_formatter(service._parse, "attribute='arg1", True))
    results.append(error_formatter(service._parse, "attribute=['arg1'", True))
    results.append(error_formatter(service._parse, "attribute=['arg1';'arg2']", True))
    results.append(error_formatter(service._parse, "attribute['arg1']", True))
    results.append(error_formatter(service._parse, "attribute=['arg1'] attribute=['arg2']", True))
    verify_all("Selenium service string errors", results, reporter=reporter)


@pytest.mark.skipif(WINDOWS, reason="ApprovalTest do not support different line feeds")
@pytest.mark.skipif(sys.version_info < (3, 12), reason="Errors change with Python 3.12")
def test_parse_service_string_errors_py3_12(service, reporter):
    results = []
    results.append(error_formatter(service._parse, "attribute=arg1", True))
    results.append(error_formatter(service._parse, "attribute='arg1", True))
    results.append(error_formatter(service._parse, "attribute=['arg1'", True))
    results.append(error_formatter(service._parse, "attribute=['arg1';'arg2']", True))
    results.append(error_formatter(service._parse, "attribute['arg1']", True))
    results.append(error_formatter(service._parse, "attribute=['arg1'] attribute=['arg2']", True))
    verify_all("Selenium service string errors", results, reporter=reporter)


@unittest.skipIf(WINDOWS, reason="ApprovalTest do not support different line feeds")
def test_split_service(service, reporter):
    results = []
    results.append(service._split("attribute='arg1'", ';'))
    results.append(service._split("attribute='arg1';attribute='arg2'", ';'))
    results.append(service._split("attribute=['arg1','arg2'];attribute='arg3'", ';'))
    results.append(service._split(" attribute = 'arg1' ; attribute = 'arg2' ", ';'))
    verify_all("Selenium service string splitting", results, reporter=reporter)


@unittest.skipIf(WINDOWS, reason="ApprovalTest do not support different line feeds")
def test_split_attribute(service, reporter):
    results = []
    results.append(service._split("attribute='arg1'", '='))
    results.append(service._split("attribute=['arg1','arg2']", '='))
    results.append(service._split(" attribute = [ 'arg1' , 'arg2' ]", '='))
    verify_all("Selenium service attribute string splitting", results, reporter=reporter)


@unittest.skipIf(WINDOWS, reason="ApprovalTest do not support different line feeds")
def test_service_create(service, reporter):
    results = []
    service_str = "service_args=['--log-level=DEBUG']"
    brwsr_service = service.create("chrome", service_str)
    results.append(brwsr_service.service_args)

    service_str = f"{service_str};service_args=['--append-log', '--readable-timestamp']"
    brwsr_service = service.create("chrome", service_str)
    results.append(brwsr_service.service_args)

    service_str = f"{service_str};service_args=['--disable-build-check']"
    brwsr_service = service.create("chrome", service_str)
    results.append(brwsr_service.service_args)

    verify_all("Selenium service", results, reporter=reporter)


@unittest.skipIf(WINDOWS, reason="ApprovalTest do not support different line feeds")
def test_importer(service, reporter):
    results = []
    results.append(service._import_service("firefox"))
    results.append(service._import_service("headless_firefox"))
    results.append(service._import_service("chrome"))
    results.append(service._import_service("headless_chrome"))
    results.append(service._import_service("ie"))
    results.append(service._import_service("edge"))
    results.append(service._import_service("safari"))
    verify_all("Selenium service import", results, reporter=reporter)


def error_formatter(method, arg, full=False):
    try:
        return method(arg)
    except Exception as error:
        if full:
            return f"{arg} {error}"
        return "{} {}".format(arg, error.__str__()[:15])