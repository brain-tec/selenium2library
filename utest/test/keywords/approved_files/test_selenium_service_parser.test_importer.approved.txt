Selenium service import

0) <class 'selenium.webdriver.firefox.service.Service'>
1) <class 'selenium.webdriver.firefox.service.Service'>
2) <class 'selenium.webdriver.chrome.service.Service'>
3) <class 'selenium.webdriver.chrome.service.Service'>
4) <class 'selenium.webdriver.ie.service.Service'>
5) <class 'selenium.webdriver.edge.service.Service'>
6) <class 'selenium.webdriver.safari.service.Service'>
